src/Order.hs

```diff
@@ -136,15 +138,35 @@ porderSuccessor foldCount orderInput orderOutput = unTermCont $ do
       (foldCount + 1)
       perror
 
+puniqueOrderedTxOuts :: Term s ((PBuiltinList (PAsData PInteger)) :--> (PInteger :--> PTxOut) :--> (PBuiltinList PTxOut))
+puniqueOrderedTxOuts =
```

Comment:
looks good, but lift the elemAt parameter into an outer lambda to avoid redundantly passing it in every recursive call.

For example:
```
pfind :: PIsListLike l a => Term s ((a :--> PBool) :--> l a :--> PMaybe a)
pfind = phoistAcyclic $ plam $ \f ->
   (pfix #$ plam $ \self xs ->
    pelimList
      ( \y ys ->
          pif
            (f # y)
            (pcon $ PJust y)
            (self # ys)
      )
      (pcon PNothing)
      xs
    )
```